### PR TITLE
Add support for colored Badge component

### PR DESCRIPTION
**Pull request by felixjung**
We need to support colored badges for various features. @majahoughton still needs to provide the correct colors for this.

### CODE REVIEW

File: src/components/Badge/Badge.js [hunk 1]
```diff
@@ -2,26 +2,99 @@ import PropTypes from 'prop-types';
...
+    [colorNames.SUCCESS]: `
+      background-color: ${theme.colors.g500};
+      color: ${theme.colors.white};
+
+      &:hover {
+        background-color: ${theme.colors.g700};
+      }
+
+      &:active {
+        background-color: ${theme.colors.y900};
```
Comment (herberthenrique) [inline src/components/Badge/Badge.js]: Should it be `g900` right?

File: src/components/Badge/Badge.js [hunk 2]
```diff
@@ -2,26 +2,99 @@ import PropTypes from 'prop-types';
...
+    [colorNames.DANGER]: `
+      background-color: ${theme.colors.r500};
+      color: ${theme.colors.white};
+
+      &:hover {
+        background-color: ${theme.colors.r700};
+      }
+
+      &:active {
+        background-color: ${theme.colors.y900};
```
Comment (connor-baer) [inline src/components/Badge/Badge.js]: Same here. r900.

File: src/components/Badge/Badge.js [hunk 3]
```diff
@@ -2,26 +2,99 @@ import PropTypes from 'prop-types';
...
+    [colorNames.PRIMARY]: `
+      background-color: ${theme.colors.b500};
+      color: ${theme.colors.white};
+
+      &:hover {
+        background-color: ${theme.colors.b700};
+      }
+
+      &:active {
+        background-color: ${theme.colors.y900};
```
Comment (connor-baer) [inline src/components/Badge/Badge.js]: And here?

**Comment by majahoughton** [comment]
@felixjung Here are the values:

Blue: 500
Green: 700
Yellow: 700
Grey: 500
Red: 500

All values work with white text

**Comment by elanoism** [comment]
@felixjung @majahoughton 

Hi, I am having a design with neutral here: https://share.goabstract.com/22949b74-a1b5-42e9-a9b4-b9856bfcaefd

Not sure if it is relevant, but wanted to mention it.

**Comment by majahoughton** [comment]
@elanoism that's right, the grey value is Neutral, made the change to the previous comment for clarification

File: src/components/Badge/Badge.js [hunk 4]
```diff
@@ -2,26 +2,99 @@ import PropTypes from 'prop-types';
 import styled, { css } from 'react-emotion';
 import { size } from 'polished';
 
+import { values } from '../../util/fp';
```
Comment (mlent) [inline src/components/Badge/Badge.js]: Sorry, I thought I read elsewhere that you and fernando agreed to use lodash for better browser support? And here you're using the custom FP functions? I'm confuzzled.

**Comment by fernandofleury** [comment]
I'm gonna pick this one up and fix the discussed topics.

**Pull request merged**